pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,389 @@
+        status_read=self.adapter.connection.read_bytes(5)
+        status=[]
+        for i in range(0,5):
+            status.append(status_read[i])
+        return status
+
+    def status_readable(self):
+        """
+        Delivers a human-readble status based on the status() command
+
+        Returns
+        -------
+        hrs : str
+            human readable status
+
+        """
+        status_read=self.status()
+        d_act=6-(status_read[0]&3)
+        m_act=list(self.MODES.keys())[((status_read[0]>>5)&7)-1]
+        if m_act=='DCV':
+            m_str='V'
+        if m_act=='ACV':
+            m_str='V'
+        if m_act=='DCI':
+            m_str='A'
+        if m_act=='ACI':
+            m_str='A'
+        if m_act=='R2W':
+            m_str='Ohm'
+        if m_act=='R4W':
+            m_str='Ohm'
+        if m_act=='Rext':
+            m_str='Ohm'
+
+        r_act=list(self.RANGES[m_act].keys())[(((status_read[0]>>2)&7)-1)]
+        if r_act <= 1:
+            r_str=str(r_act*1000)+' m'
+        if r_act >=1E6:
+            r_str=str(r_act/1E6)+' M'
+        if r_act >=1000:
+            r_str=str(r_act/1000)+' k'
+        else:
+            r_str=str(r_act)+ ' '
+        hrs = '----- HP3478 ----- Status -----\n'
+        hrs = hrs + 'The current range is '+r_str+m_str+' with '+str(d_act)+' 1/2 digits\r\n'
+
+        hrs = hrs + 'General status: \r\n'
+        hrs = hrs + '     Internal trigger: '+str(status_read[1]&1)+'\r\n'
+        hrs = hrs + '     Autorange: '+str(status_read[1]>>1&1)+'\r\n'
+        hrs = hrs + '     Auto Zero: '+str(status_read[1]>>2&1)+'\r\n'
+        hrs = hrs + '     50 Hz opereration: '+str(status_read[1]>>3&1)+'\r\n'
+        hrs = hrs + '     Front/Rear: '+str(status_read[1]>>4&1)+'\r\n'
+        hrs = hrs + '     CalibrationRAM en: '+str(status_read[1]>>6&1)+'\r\n'
+        hrs = hrs + '     External trigger: '+str(status_read[1]>>7&1)+'\r\n'
+        hrs = hrs + 'Srq mask: '+str(status_read[2])+'\r\n'
+        hrs = hrs + 'Error status: '+str(status_read[3])+'\r\n'
+        hrs = hrs + 'DAC value: '+str(status_read[4])+'\r\n'
+        return hrs
+
+    @property
+    def auto_zero(self):
+        """
+        Checks the autozero settings on the HP3478
+
+        Returns
+        -------
+        autozero : int
+            0 == Autozero off
+            1 == Autozero on
+
+        """
+        status_read=self.status()
+        autozero=status_read[1]>>2&1
+        return autozero
+
+    @auto_zero.setter
+    def auto_zero(self,on_off):
+        """
+        Offers the possibility to enable or disable the Autozero feature
+
+        Parameters
+        ----------
+        on_off : int
+            expects either 0 or 1
+            0 --> disabled autozero
+            1 --> enables autozero.
+
+        Returns
+        -------
+        None.
+
+        """
+        if on_off==0:
+            self.write('Z0')
+        else:
+            if on_off==1:
+                self.write('Z1')
+
```

Comment:
What about self.write('Z{}'.format(on_off)) ?

Reply:
That was a good hint!
Thank you very much!